etc/bash_completion.d/dnf-completion.bash

```diff
@@ -71,106 +74,39 @@ _dnf()
     if [[ $command ]]; then
 
         case $command in
-            install|update|info)
-                if [[ "$cur" == \.* ]] || [[ "$cur" == \/* ]]; then
-                    [[ $command != "info" ]] && ext='@(rpm)' || ext=''
-                else
-                    if [ -r $cache_file ]; then
-                        COMPREPLY=( $( compgen -W '$( sqlite3 $cache_file "select pkg from available WHERE pkg LIKE \"$cur%\"" )' ) )
-                    else
-                        COMPREPLY=( $( compgen -W '$( python << END
-import dnf
-import os
-import logging
-class NullHandler(logging.Handler):
-    def emit(self, record):
-        pass
-h = NullHandler()
-logging.getLogger("dnf").addHandler(h)
-b = dnf.Base()
-b.read_all_repos()
-if not dnf.util.am_i_root():
-    cachedir = dnf.yum.misc.getCacheDir()
-    b.conf.cachedir = cachedir
-b.conf.substitutions["releasever"] = dnf.rpm.detect_releasever("/")
-suffix = dnf.conf.parser.substitute(dnf.const.CACHEDIR_SUFFIX, b.conf.substitutions)
-for repo in b.repos.values():
-    repo.basecachedir = os.path.join(b.conf.cachedir, suffix)
-    repo.md_only_cached = True
-try:
-    b.fill_sack()
-except dnf.exceptions.RepoError:
-    pass
-q = b.sack.query().available()
-for pkg in q:
-    print("{}.{}").format(pkg.name, pkg.arch)
-END
-)' -- "$cur" ) )
-                    fi
+            install|update|upgrade)
+                if [[ "$cur" != \.* ]] && [[ "$cur" != \/* ]] && [[ "$cur" != ~* ]]; then
+                    _dnf_helper $command "$cur"
+                    # FIXME: fix sqlitedb to store hawkey.Package
```

Comment:
How you wanna store hawkey.Package object? I doubt it would work even with pickle when the main part is in C Python.